src/main/java/com/microsoft/sqlserver/jdbc/SQLServerCallableStatement.java

```diff
@@ -150,6 +151,22 @@ public void registerOutParameter(int index, int sqlType) throws SQLServerExcepti
             case microsoft.sql.Types.DATETIMEOFFSET:
                 param.setOutScale(7);
                 break;
+            case java.sql.Types.DECIMAL:
+                // Dynamically handle the scale for DECIMAL output parameters.
+                // The scale for the DECIMAL type is fetched from the ParameterMetaData.
+                // This provides flexibility to automatically apply the correct scale as per the database metadata.
+                ParameterMetaData parameterMetaData = this.getParameterMetaData();
```

Comment:
I'm pretty sure this will incur a round trip to the database to get the metadata. That would be a significant performance hit if so. You can validate by running the SQL Server Profiler, watching executions against the DB while exercising this code. We would not want to add a round trip to the server for this.